backtrace-library/src/androidTest/java/backtraceio/library/BacktraceClientMetricsTest.java

```diff
@@ -0,0 +1,938 @@
+
+@RunWith(AndroidJUnit4.class)
+public class BacktraceClientMetricsTest {
+    public Context context;
+    public BacktraceClient backtraceClient;
+    public BacktraceCredentials credentials;
+    private final String summedEventName = "activity-changed";
+    // existing attribute name in Backtrace
+    private final String[] uniqueAttributeName = {"uname.version", "cpu.boottime", "screen.orientation", "battery.state", "device.airplane_mode", "device.sdk", "device.brand", "system.memory.total", "uname.sysname", "application.package"};
+
+    private final String defaultBaseUrl = BacktraceMetrics.defaultBaseUrl;
+    private final String token = "aaaaabbbbbccccf82668682e69f59b38e0a853bed941e08e85f4bf5eb2c5458";
+    private final String universeName = "testing-universe-name";
+    private final String _uniqueEventsSubmissionUrl = "unique-events/submit?token=" + token + "&universe=" + universeName;
+    private final String _summedEventsSubmissionUrl = "summed-events/submit?token=" + token + "&universe=" + universeName;
+
+    static {
+        System.loadLibrary("backtrace-native");
+    }
+
+    /**
+     * NOTE: Some of these tests are very time-sensitive so you may occasionally get false negative results.
+     * For best results run under low CPU load and low memory utilization conditions.
+     */
+
+    @Before
+    public void setUp() {
+        context = InstrumentationRegistry.getContext();
+        credentials = new BacktraceCredentials("https://example-endpoint.com/", "");
+        BacktraceDatabase database = new BacktraceDatabase(context, context.getFilesDir().getAbsolutePath());
+
+        backtraceClient = new BacktraceClient(context, credentials, database);
+
+        BacktraceLogger.setLevel(LogLevel.DEBUG);
+    }
+
+    public class MockRequestHandler implements EventsRequestHandler {
+        public int numAttempts = 0;
+        public int statusCode = 200;
+        public String lastEventPayloadJson;
+
+        @Override
+        public EventsResult onRequest(EventsPayload data) {
+            String eventsPayloadJson = BacktraceSerializeHelper.toJson(data);
+            lastEventPayloadJson = eventsPayloadJson;
+            numAttempts++;
+
+            BacktraceResultStatus status;
+            if (statusCode == 200) {
+                status = BacktraceResultStatus.Ok;
+            } else {
+                status = BacktraceResultStatus.ServerError;
+            }
+            return new EventsResult(data, eventsPayloadJson, status, statusCode);
+        }
+    }
+
+    @Test
+    public void disableAutoSend() {
+        backtraceClient.metrics.enable(new BacktraceMetricsSettings(universeName, token, defaultBaseUrl, 0));
+        MockRequestHandler mockRequestHandler = new MockRequestHandler();
+        backtraceClient.metrics.setUniqueEventsRequestHandler(mockRequestHandler);
+
+        backtraceClient.metrics.addUniqueEvent(uniqueAttributeName[0]);
+        try {
```

Comment:
wouldn't it be better if we will call directly a method responsible for sending data that the timer executes in the backtracemetrics class implementation? what will give us sleep? do we want to check events every sec?  what if sleep timer changed in the future for some reason? This test will give us false positive information.

Reply:
You're right, I don't think this timer adds anything. I just want to make sure we DON'T send something but timer won't help here. I'll just remove the timer. Data sending timer won't help, we specifically disable data sending timer for this test.